### ISSUE TITLE
Prevent creeps from attacking structures

### ISSUE THREAD

**Issue by nvs** [nvs/gem#244]
Given that block checking upon placement in #132 is not planned soon, it may be best to improve the current attack detection behavior. Ideally, if the creeps do attack a player owned structure they should be teleported immediately to another location. The downside is that a block may not be discovered immediately. But the biggest pro is that a player's maze may not become incapacitated.

**Comment by nvs** [comment]
Blocked by #132.

**Comment by nvs** [comment]
This is basically a duplicate of #373 (or, rather, that was a duplicate of this).
